build/src/src/modules/sshPassword.js

```
@@ -0,0 +1,86 @@
+ * - To change the password
+ * `docker run --rm -v /etc:/etc --privileged --entrypoint="" dappmanager.dnp.dappnode.eth:0.2.5 sh -c "echo dappnode:newpassword | chpasswd"`
+ *   The new password is provided after `dappnode:` in this case it's `newpassword`
+ *   This command will return `chpasswd: password for 'dappnode' changed` if all went well
+ *   image = "dappmanager.dnp.dappnode.eth:0.2.5" (or whatever the last version is)
+ *
+ */
+
+const baseCommand = `docker run --rm -v /etc:/etc --privileged --entrypoint=""`;
+
+// If the DAPPMANAGER image changes, this node app MUST be reseted
+let cacheDappmanagerImage;
+async function getDappmanagerImage() {
+  if (cacheDappmanagerImage) return cacheDappmanagerImage;
+  const res = await shell(
+    `docker ps --filter "name=dappmanager.dnp.dappnode.eth" --format "{{.Image}}"`
+  );
+  if (!res) throw Error("No image found for dappmanager.dnp.dappnode.eth");
+  const dappmanagerImage = res.trim();
+  cacheDappmanagerImage = dappmanagerImage;
+  return dappmanagerImage;
+}
+
+/**
+ * Checks if the SSH password of the host for the user `dappnode`
+ * is NOT the insecure default set at installation time.
+ * It does so by checking if the current salt is `insecur3`
+ *
+ * @returns {bool} true = is secure / false = is not
+ */
+async function isSshPasswordSecure() {
+  const image = await getDappmanagerImage();
+  const res = await shell(
+    `${baseCommand} ${image} sh -c "grep dappnode:.*${insecureSalt} /etc/shadow"`
+  );
+  return !res;
+}
+
+/**
+ * Changes the SSH password of the host for the user `dappnode`
+ * Only allows it if the current password has the salt `insecur3`
+ *
+ * @param {string} newPassword super-secure-password
+ */
+async function changeSshPassword(newPassword) {
+  if (!newPassword) throw Error("newPassword must be defined");
+
+  if (await isSshPasswordSecure())
+    throw Error(
+      "The SSH password can only be changed if it's the insecure default"
+    );
```

I think we should return an error if the password is shorter than 8 chars

---

Ok